torch/onnx/symbolic_opset16.py

```diff
@@ -75,12 +75,47 @@ def scatter_add(g: jit_utils.GraphContext, self, dim, index, src):
     src_sizes = symbolic_helper._get_tensor_sizes(src)
     index_sizes = symbolic_helper._get_tensor_sizes(index)
 
-    if src_sizes != index_sizes:
+    if len(src_sizes) != len(index_sizes):
         return symbolic_helper._unimplemented(
             "scatter_add",
             f"`index` ({index_sizes}) should have the same dimensionality as `src` ({src_sizes})",
         )
 
+    if src_sizes != index_sizes:
+        # In ONNX, src and index are required to be the same rank and shape
+        # However, in PyTorch, src is only required to have the same rank as index,
+        # and shape would be accomodated. In static shape, converter can apply Slice op
+        # to accommodate. We use Slice to adjust to shape of src if it's not the same
+        # as index.
+        # More detail on: https://github.com/onnx/onnx/issues/4672
+        axes = list()
+        ends = list()
+        for idx, d in enumerate(index_sizes):
+            if d is None:
+                # the axe with dynamic shape is ignored, and will be aligned by
+                # setType later
+                continue
+            if src_sizes[idx] < d:
+                return symbolic_helper._unimplemented(
+                    "scatter_add",
+                    f"`index` ({index_sizes}) should have smaller or equal (<=) size at any dimension than `src` ({src_sizes})",
+                )
+            axes.append(idx)
+            ends.append(d)
+        starts = [0] * len(ends)
+        if axes and starts and ends:
+            src = symbolic_helper._slice_helper(
+                g, src, axes=axes, starts=starts, ends=ends
+            )
+        # Align the dynamic sizes of src and index
+        # NOTE: Even if users set src and index with different dynamic axes, they are
```

Comment:
It looks like symbolic function does not have enough information to remediate if shape don't match under dynamic size. 
Does setting sizes manually below has any effect? (Is it to pass onnx checker?) If not I'd suggest remove it and comment above why mismatch under dynamic shape cannot be fixed here.

Reply:
Done